PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+     an entitlement shared with them, but might not be entirely aware of the benefits they now have.
+     */
+    public let ownershipType: PurchaseOwnershipType
+
+    // TODO(post-migration): Make this internal
+    // TODO(cleanup): Codable
+    public init(entitlementId: String,
+                entitlementData: [String: Any],
+                productData: [String: Any],
+                dateFormatter: DateFormatter,
+                requestDate: Date?) {
+        // Entitlement data
+        let entitlementExpiresDateString = entitlementData["expires_date"] as? String
+        let entitlementPurchaseDateString = entitlementData["purchase_date"] as? String
+        let productIdString = entitlementData["product_identifier"] as? String
+
+        // Product data
+        let periodTypeString = productData["period_type"] as? String
+        let originalPurchaseDateString = productData["original_purchase_date"] as? String
+        let productExpiresDateString = productData["expires_date"] as? String
+        let storeString = productData["store"] as? String
+        let maybeSandbox = productData["is_sandbox"] as? NSObject // This could be a String or NSNumber
+        let unsubscribeDetectedAtString = productData["unsubscribe_detected_at"] as? String
+        let billingIssuesDetectedAtString = productData["billing_issues_detected_at"] as? String
+        let ownershipType = productData["ownership_type"] as? String
+
+        let store = Self.parseStore(store: storeString)
+        let expirationDate = Self.parseDate(dateString: productExpiresDateString, withDateFormatter: dateFormatter)
+        let unsubscribeDetectedAt = Self.parseDate(dateString: unsubscribeDetectedAtString,
+                                                   withDateFormatter: dateFormatter)
+        let billingIssueDetectedAt = Self.parseDate(dateString: billingIssuesDetectedAtString,
+                                                    withDateFormatter: dateFormatter)
+        let entitlementExpiresDate = Self.parseDate(dateString: entitlementExpiresDateString,
+                                                    withDateFormatter: dateFormatter)
+
+        self.store = store
+        self.expirationDate = expirationDate
+        self.unsubscribeDetectedAt = unsubscribeDetectedAt
+        self.billingIssueDetectedAt = billingIssueDetectedAt
+        self.identifier = entitlementId
+        self.productIdentifier = productIdString!
+
+        let isSandbox: Bool
+        if maybeSandbox?.responds(to: #selector(getter: NSNumber.boolValue)) ?? false,
+           let unwrapped = maybeSandbox as? NSNumber {
+            isSandbox = unwrapped.boolValue
+        } else if maybeSandbox?.responds(to: #selector(getter: NSString.boolValue)) ?? false,
+                  let unwrapped = maybeSandbox as? NSString {
+            isSandbox = unwrapped.boolValue
+        } else {
+            isSandbox = false
+        }
```

Comment:
although I love how thoughtful this translation is, and it is absolutely correct, I think we should just figure out which type it is so that we can remove the complexity here, since it's a value that is under our control (if this was coming from a third-party API, or something outside our control, I'd advocate for keeping this logic)

Reply:
Will update since we can see it's definitely returned as a `bool` and not a string 🎉